src/UCTNodePointer.cpp

```diff
@@ -26,17 +26,18 @@
 #include "UCTNode.h"
 
 UCTNodePointer::~UCTNodePointer() {
-    if (is_inflated()) {
-        delete read_ptr();
+    auto v = std::atomic_exchange(&m_data, INVALID);
```

Comment:
Do you need the exchange here? We can't be running this with multiple threads accessing it anyway.

Reply:
Not really, but no harm either.  We can change it to non-exchange if you insist.

Reply:
Simple assignment?